### DISCUSSION TITLE
Quality control in rhino apps?

### DISCUSSION THREAD

**Discussion by shahronak47** [Appsilon/rhino#512] [Q&A]
Hi, I have recently started using `rhino` for my shiny apps and it looks interesting. Previously, I used to structure my apps in a package format mostly using `golem`. What it did provide me was `devtools::check()`, `devtools::document()` and `devtools::test()`. This gives me additional confidence on the code that I have written and produces documentation. 

How can I achieve the same with Rhino framework?

**Comment by jakubnowicki** [comment]
As Rhino applications are, by design, not R packages, the standard checks do not apply here. Instead, a set of functions is available to ensure the quality of the application:
- `rhino::test_r` will trigger unit tests
- `rhino::test_e2e` will run front-end tests
- `rhino::lint_r`, `rhino::lint_js`, and `rhino::lint_sass` will check R, JavaScript, and Sass code using linters

**Comment by mmmmtoasty19** [comment]
Old post but I stumbled too it due to the use of `devtools::document` how can we use this functionality within a rhino app?
